cassdegrees/ui/views.py

```diff
@@ -95,3 +95,86 @@ def sampleform(request):
 
 def create_subplan(request):
     return render(request, 'createsubplan.html')
+
+
+# inspired by the samepleform function created by Daniel Jang
+def manage_courses(request):
+    # Reads the 'action' attribute from the url (i.e. manage/?action=Add) and determines the submission method
+    action = request.GET.get('action', 'Add')
+    if   action == 'Add'   : method = 'post'
+    elif action == 'Edit'  : method = 'patch'
+    elif action == 'Delete': method = 'delete'
+    else:
+         action =  'Add'   ; method = 'post'
+
+    action = 'Add'; method = 'post' # TODO Remove once other methods are implemented
+
+    courses = requests.get(request.build_absolute_uri('/api/model/course/?format=json')).json()
+    courses = [{'code': course} for course in  set([x['code'] for x in courses])]
+    # If POST request, redirect the received information to the backend:
+    render_properties = {
+        'msg': None,
+        'is_error': False
+    }
+    if request.method == 'POST':
+        # hard coded url; only temporary
+        model_api_url = request.build_absolute_uri('/api/model/course/')
+        post_data = request.POST
+        actual_request = post_data.get('_method')
+
+        # This method of transferring data to the API was inspired by:
+        # https://stackoverflow.com/questions/11663945/calling-a-rest-api-from-django-view
+        if actual_request == "post":
```

Comment:
We shouldn't be using this for these forms - we only need POST and GET. Get rid of this if statement entirely IMO.

Reply:
actual_request is now deleted.